test/garbage_collection_test.js

```diff
@@ -26,7 +26,12 @@ suite('garbage collection tests', function () {
       delayFactor: 15 * 1000,
       randomizationFactor: 0.25
     },
-    log: logger()
+    log: logger(),
+    stats: {
+      timeGen: async function (series, fn) {
+        return await fn;
```

Comment:
Right, this would just be a way to catch cases where timeGen is handed a second argument that is not a Promise -- `timeGen('some-metric', someFunction(..))` looks legitimate, but if someFunction isn't async, some-metric will always be zero even if someFunction is slow.